lib/validate.js

```diff
@@ -135,13 +142,19 @@ module.exports = {
       this.options.verbose && this.serverless.cli.log('Using multi-compile (individual packaging)');
       this.multiCompile = true;
 
+      if (this.webpackConfig.entry && !_.isEqual(this.webpackConfig.entry, entries)) {
+        throw new this.serverless.classes
```

Comment:
We should reject here instead of throwing (`return BbPromise.reject(...)`). Validate is expected to return a promise. Imo mixing exceptions and promises is not very clean. BTW: I know that this is also done in Serverless and maybe there are some remnants in this plugin too ;-)

In the unit test you should then just use `expect(validate()).to.be.rejectedWith(/.../)`